okhttp/src/main/java/io/grpc/okhttp/OkHttpServerBuilder.java

```diff
@@ -0,0 +1,376 @@
+   */
+  public static OkHttpServerBuilder forPort(SocketAddress address, ServerCredentials creds) {
+    HandshakerSocketFactoryResult result = handshakerSocketFactoryFrom(creds);
+    if (result.error != null) {
+      throw new IllegalArgumentException(result.error);
+    }
+    return new OkHttpServerBuilder(address, result.factory);
+  }
+
+  final ServerImplBuilder serverImplBuilder = new ServerImplBuilder(this::buildTransportServers);
+  final SocketAddress listenAddress;
+  final HandshakerSocketFactory handshakerSocketFactory;
+  TransportTracer.Factory transportTracerFactory = TransportTracer.getDefaultFactory();
+
+  ObjectPool<Executor> transportExecutorPool = DEFAULT_TRANSPORT_EXECUTOR_POOL;
+  ObjectPool<ScheduledExecutorService> scheduledExecutorServicePool =
+      SharedResourcePool.forResource(GrpcUtil.TIMER_SERVICE);
+
+  ServerSocketFactory socketFactory = ServerSocketFactory.getDefault();
+  long keepAliveTimeNanos = KEEPALIVE_TIME_NANOS_DISABLED;
+  long keepAliveTimeoutNanos = DEFAULT_KEEPALIVE_TIMEOUT_NANOS;
+  int flowControlWindow = DEFAULT_FLOW_CONTROL_WINDOW;
+  int maxInboundMetadataSize = GrpcUtil.DEFAULT_MAX_HEADER_LIST_SIZE;
+  int maxInboundMessageSize = GrpcUtil.DEFAULT_MAX_MESSAGE_SIZE;
+
+  private OkHttpServerBuilder(
+      SocketAddress address, HandshakerSocketFactory handshakerSocketFactory) {
+    this.listenAddress = Preconditions.checkNotNull(address, "address");
+    this.handshakerSocketFactory =
+        Preconditions.checkNotNull(handshakerSocketFactory, "handshakerSocketFactory");
+  }
+
+  @Internal
+  @Override
+  protected ServerBuilder<?> delegate() {
+    return serverImplBuilder;
+  }
+
+  @VisibleForTesting
+  OkHttpServerBuilder setTransportTracerFactory(TransportTracer.Factory transportTracerFactory) {
+    this.transportTracerFactory = transportTracerFactory;
+    return this;
+  }
+
+  /**
+   * Override the default executor necessary for internal transport use.
+   *
+   * <p>The channel does not take ownership of the given executor. It is the caller' responsibility
+   * to shutdown the executor when appropriate.
+   */
+  public OkHttpServerBuilder transportExecutor(Executor transportExecutor) {
+    if (transportExecutor == null) {
+      this.transportExecutorPool = DEFAULT_TRANSPORT_EXECUTOR_POOL;
+    } else {
+      this.transportExecutorPool = new FixedObjectPool<>(transportExecutor);
+    }
+    return this;
+  }
+
+  /**
+   * Override the default {@link ServerSocketFactory} used to listen. If the socket factory is not
+   * set or set to null, a default one will be used.
+   */
+  public OkHttpServerBuilder socketFactory(ServerSocketFactory socketFactory) {
+    if (socketFactory == null) {
+      this.socketFactory = ServerSocketFactory.getDefault();
+    } else {
+      this.socketFactory = socketFactory;
+    }
+    return this;
+  }
+
+  /**
+   * Sets a custom keepalive time, the delay time for sending next keepalive ping after a period of
```

Comment:
Also mention that negative values will be rejected?

Reply:
`io.grpc.ServerBuilder` would be the important place to change that. This is old text from NettyServerBuilder. We're going to do a review of it this Thursday. In this PR I still need to merge with https://github.com/grpc/grpc-java/pull/9176 so I can use `@Override` and such here.